### ISSUE TITLE
Ember Asset loader does not take rootURL into consideration

### ISSUE THREAD

**Issue by ghost** [ember-engines/ember-asset-loader#29]
Assets are loaded absolutely instead of relative to account for a rootURL setting

**Comment by trentmwillis** [comment]
Yep, an oversight in the implementation. Everything we've tested against was at root.

**Comment by ghost** [comment]
Thanks much... I'll see if I can debug and submit a PR for this.... I've tried to so far but running in to competing priorities right now

**Comment by mattmarcum** [comment]
I just ran into this bug too.  ~~I thought maybe implementing this would fix it: https://github.com/ember-engines/ember-asset-loader#generating-custom-uris  but it doesn't~~

Actually, using `generateURI` does work:

```
const config = require('./tests/dummy/config/environment')();

module.exports = function(defaults) {
  let app = new EmberAddon(defaults, {
    // Add options here
    assetLoader: {
      generateURI: function(filePath) {
        return `${config.rootURL}${filePath}`;
      }
    }
  });
```

Is that the best way to get the config - just require and instantiate it?  I tried looking for it in the `defaults` param but couldn't find it in there.

**Comment by villander** [comment]
@rwjblue can you close this issue, please? this approach has gone stale - https://github.com/ember-engines/ember-asset-loader/pull/33#issuecomment-294009003

**Comment by dfreeman** [comment]
Should this have been closed? The proposed solution in #33 was stale, but the issue itself still remains, doesn't it?

**Comment by dbouwman** [comment]
For others who land here... earlier this week, I used `generateURI` in my host app's `ember-cli-build.js` to inform the host app that the engines should be loaded from our CDN...

We use `broccoli-asset-rev` along with `fingerprintOptions.prepend` and this is what the function looks like:
```
// host-app's ember-cli-build.js
const app = new EmberApp(defaults, {
    assetLoader: {
      generateURI: function (filePath) {
        // if we have specified something to prepend, and the file is in engines-dist...
        if (fingerprintOptions.prepend && filePath.indexOf('engines-dist') > -1) {
          return `${fingerprintOptions.prepend}${filePath.substring(1)}`;
        } else {
          return filePath;
        }
      }
    },
}
```
This ensures that only files in `engines-dist` are impacted, and only when `fingerprintOptions.prepend` is defined. Hope this is useful...
